packages/typespec-ts/test/modularUnit/anonymousModel.spec.ts

```diff
@@ -0,0 +1,973 @@
+      });
+    });
+    describe("happens as a property in response body", async () => {
+      it("should map empty anonymous  model({}) => Record<string, any> & empty named model(A {}) => Record<string, any>", async () => {
+        const tspContent = `
+        model EmptyModel {
+        }
+        model ReturnBody {
+          emptyAnomyous: {};
+          emptyAnomyousArray: {}[];
+          emptyAnomyousDict: Record<{}>;
+          emptyModel: EmptyModel;
+          emptyModelArray: EmptyModel[];
+          emptyModelDict: Record<EmptyModel>;
+        }
+        op read(): ReturnBody;
+        `;
+        const modelFile = await emitModularModelsFromTypeSpec(tspContent);
+        // console.log(modelFile?.getFullText());
+        assertEqualContent(
+          modelFile?.getFullText()!,
+          `
+        export interface ReturnBody {
+          emptyAnomyous: Record<string, any>;
+          emptyAnomyousArray: Record<string, any>[];
+          emptyAnomyousDict: Record<string, Record<string, any>>;
+          emptyModel: EmptyModel;
+          emptyModelArray: EmptyModel[];
+          emptyModelDict: Record<string, EmptyModel>;
+        }
+
+        export interface EmptyModel {}
+        `
+        );
+        const operationFiles = await emitModularOperationsFromTypeSpec(
+          tspContent
+        );
+        assert.equal(operationFiles?.length, 1);
+        assertEqualContent(
+          operationFiles?.[0]?.getFullText()!,
+          `
+        import { TestingContext as Client } from "../rest/index.js";
+        import {
+          StreamableMethod,
+          operationOptionsToRequestParameters,
+        } from "@azure-rest/core-client";
+
+        export function _readSend(
+          context: Client,
+          options: ReadOptions = { requestOptions: {} }
+        ): StreamableMethod<Read200Response> {
+          return context
+            .path("/")
+            .get({ ...operationOptionsToRequestParameters(options) });
+        }
+
+        export async function _readDeserialize(
+          result: Read200Response
+        ): Promise<ReturnBody> {
+          if (result.status !== "200") {
+            throw result.body;
+          }
+
+          return {
+            emptyAnomyous: result.body["emptyAnomyous"],
+            emptyAnomyousArray: result.body["emptyAnomyousArray"],
+            emptyAnomyousDict: result.body["emptyAnomyousDict"],
+            emptyModel: {},
+            emptyModelArray: (result.body["emptyModelArray"] ?? []).map(() => ({})),
+            emptyModelDict: result.body["emptyModelDict"],
+          };
+        }
+
+        export async function read(
+          context: Client,
+          options: ReadOptions = { requestOptions: {} }
+        ): Promise<ReturnBody> {
+          const result = await _readSend(context, options);
+          return _readDeserialize(result);
+        }`
+        );
+      });
+
+      it("should map non-empty anonymous  model({ ... }) => { ... }", async () => {
```

Comment:
I wonder if we should add some ut for non-empty anonymous model like above empty model's case ?
          nonemptyAnomyous: {...};
          nonemptyAnomyousArray: {...}[];
          nonemptyAnomyousDict: Record<{...}>;

Reply:
updated